docarray/index/abstract.py

```diff
@@ -706,6 +707,21 @@ def _filter_by_parent_id(self, id: str) -> Optional[List[str]]:
     # These might be useful in your subclass implementation  #
     ##########################################################
 
+    @staticmethod
+    def custom_issubclass(x, A_tuple) -> bool:
+        """
+        This is a modified version of the built-in 'issubclass' function to support non-class input.
+        Traditional 'issubclass' calls can result in a crash if the input is list/tuple.
+
+        :param x: A class 'x'
+        :param A_tuple: A class, or a tuple of classes.
+        :return: A boolean value - 'True' if 'x' is a subclass of 'A_tuple', 'False' otherwise.
+                 Note that if the origin of 'x' is a list or tuple, the function immediately returns 'False'.
+        """
+        if get_origin(x) in (list, tuple):
```

Comment:
What about other origins? I think Dict for example applies in the same way.
Also, are there other things that aren't classes (= break normal issubclass) but also don't have an origin? TypeVars for example? Would be nice to cover as many cases as possible here.